src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -565,7 +565,7 @@ private static size_t CurlReceiveHeadersCallback(IntPtr buffer, size_t size, siz
                             if (colonIndex > 0)
                             {
                                 string headerName = responseHeader.Substring(0, colonIndex);
-                                string headerValue = responseHeader.Substring(colonIndex + 1);
+                                string headerValue = responseHeader.Substring(colonIndex + 1).Trim();
```

Comment:
If @davidsh says this needs to be fixed, I will handle it in the other PR. As you pointed out, CurlHandler and WinHttpHandler are in sync with ```Trim()```. 

On a side note, may be, we should consider moving the header parsing to a common place.